packages/uni_ui/lib/calendar/calendar_item.dart

```diff
@@ -0,0 +1,160 @@
+import 'package:figma_squircle/figma_squircle.dart';
+import 'package:flutter/material.dart';
+
+class _CalendarItemDate extends StatelessWidget {
+  const _CalendarItemDate({
+    required this.eventPeriod,
+  });
+
+  final DateTimeRange? eventPeriod;
+
+  @override
+  Widget build(BuildContext context) {
+    if (eventPeriod != null) {
+      return Column(
+        children: [
+          Text(
+            parsePeriod(eventPeriod!),
+            style: TextStyle(
+              fontSize: 15,
+              height: 1,
+              fontWeight: FontWeight.w500,
+            ),
+          ),
+          Text(
+            eventPeriod!.end.year.toString(),
+            style: TextStyle(
+              color: Theme.of(context).colorScheme.outline,
+              fontSize: 11,
+            ),
+          ),
+        ],
+      );
+    } else {
+      return Padding(
+        padding: EdgeInsets.only(top: 10),
+        child: Text(
+          "TBD",
+          style: TextStyle(
+            fontSize: 15,
+            fontWeight: FontWeight.w500,
+          ),
+        ),
+      );
+    }
+  }
+
+  static String monthToString(int month) {
+    // TODO: Support Portuguese
+    const strMonths = [
+      "Jan",
+      "Feb",
+      "Mar",
+      "Apr",
+      "May",
+      "Jun",
+      "Jul",
+      "Aug",
+      "Sep",
+      "Oct",
+      "Nov",
+      "Dec"
+    ];
+    return strMonths[month - 1];
+  }
+
+  static String parsePeriod(DateTimeRange period) {
+    final start = period.start, end = period.end;
+
+    if (start.month == end.month) {
+      return start.day == end.day
+          ? "${start.day} ${monthToString(start.month)}"
+          : "${start.day} - ${end.day} ${monthToString(start.month)}";
+    } else {
+      return "${start.day} ${monthToString(start.month)} - ${end.day} ${monthToString(end.month)}";
+    }
+  }
+}
+
+class CalendarItem extends StatelessWidget {
+  const CalendarItem({
+    super.key,
+    required this.eventName,
+    this.eventPeriod,
```

Comment:
I think event period must be required here. Is there any event that does not have a date?